lib/song.rb

```diff
@@ -0,0 +1,58 @@
+class Song
+  extend Concerns::Findable
+
+  attr_accessor :name
+  attr_reader :artist
+  attr_reader :genre
+
+  @@all = []
+
+  def initialize(name, artist = nil, genre = nil)
+    @name = name
+    self.artist = artist
+    self.genre = genre
+  end
+
+  def artist=(artist)
+    @artist = artist
+    artist && artist.add_song(self)
+  end
+
+  def genre=(genre)
+    @genre = genre
+    genre && (genre.songs << self unless genre.songs.include?(self))
+  end
+
+  def self.all
+    @@all
+  end
+
+  def save
+    @@all << self
+    self
+  end
+
+  def self.destroy_all
+    @@all.clear
+  end
+
+  def self.create(name, artist = nil, genre = nil)
+    Song.new(name, artist, genre).save
+  end
+
```

Comment:
#### Code Quality - Critical
```self.all``` ```save``` ```self.destroy_all``` and ```self.create``` can be refactored into a parent class. This way they are not repeated in the ```Artist``` and ```Genre``` classes.